pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -517,28 +518,37 @@ func (t *CycleNodeRequestTransitioner) transitionHealing() (reconcile.Result, er
 		return t.transitionToFailed(err)
 	}
 
-	// try and re-attach the nodes, if any were un-attached
 	for _, node := range t.cycleNodeRequest.Status.NodesToTerminate {
+		// nodes in NodesToTerminate may have been terminated, so check if they still exist
+		if nodeExists, err := k8s.NodeExists(node.Name, t.rm.RawClient); err != nil || !nodeExists {
```

Comment:
You need to catch the error from here. In the function it already checks if the error returned indicated that the node no longer exists which means that legitimate api errors will be returned here. Those need to be caught.